### ISSUE TITLE
Remove cells filter alters input dataset

### ISSUE THREAD

**Issue by banesullivan** [pyvista/pyvista#1855]
**Describe the bug, what's wrong, and what you expect:**

The `remove_cells` filter seems to be removing cells inplace on the source mesh and returning a new mesh with the cells removed as well. This is very unexpected -- the source mesh should be left unnafected!

-----

**To Reproduce**

I uncovered this when doing https://github.com/pyvista/pyvista/discussions/1815#discussioncomment-1703846

```py
from pyvista import examples

truth = examples.load_hexbeam()
source = truth.copy()
out = source.remove_cells([0])

p = pv.Plotter(notebook=0, shape=(1,2))
p.add_mesh(truth.extract_all_edges(), color='k', line_width=10)
p.add_mesh(source, color='tan')
p.subplot(0,1)
p.add_mesh(truth.extract_all_edges(), color='k', line_width=10)
p.add_mesh(out, color='tan')
p.link_views()
p.show()
```
<img width="624" alt="Screen Shot 2021-11-25 at 11 54 51 PM" src="https://user-images.githubusercontent.com/22067021/143539477-e99999b5-1d39-4265-b7b6-2adf34f38e1e.png">

**Comment by tkoyama010** [comment]
This is the problem of the default parameter of in_place argument. I am +0.5 for change the default value. The issue of backward compatibility needs to be discussed.

**Comment by banesullivan** [comment]
Well that presents another problem as well. If `in_place=True` for a filter it should not have any return vale

**Comment by banesullivan** [comment]
IMO, no filters should have a default value of `True` for `inplace`. `inplace` should *always* be opt-in rather than opt-out because of its ambiguity / difficulty in tracking what happened to your data

**Comment by banesullivan** [comment]
For example, this should not error, but it does because the filter is editing in place by default.

```py
box = pv.Box()

left = box.remove_cells([0,1,2])
right = box.remove_cells([3,4,5])
```
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
/var/folders/2x/4df66vz13pv4hj_ysv23tyyc0000gn/T/ipykernel_85626/2585597499.py in <module>
      2 
      3 left = box.remove_cells([0,1,2])
----> 4 right = box.remove_cells([3,4,5])

~/Software/pyvista/pyvista/pyvista/core/pointset.py in remove_cells(self, ind, inplace)
    110                                  f'number of cells ({self.n_cells}')
    111         ghost_cells = np.zeros(self.n_cells, np.uint8)
--> 112         ghost_cells[ind] = _vtk.vtkDataSetAttributes.DUPLICATECELL
    113 
    114         if inplace:

IndexError: index 3 is out of bounds for axis 0 with size 3
```

The typical PyVista syntax is:
```
result = source.filter(options)
```
There have been previous discussions that established this convention and the `remove_cells` filter does not adhere by editing the `source` AND producing `result`

**Comment by banesullivan** [comment]
From my perspective, the current default value of `inplace=True` for this filter is a bug.

**Comment by tkoyama010** [comment]
Sorry for my misunderstanding. Now I am +1 for fix `inplace` defalut value to `True`.

**Comment by banesullivan** [comment]
So I started drafting a branch to address this and oh my.... the usage of `inplace` has gone off the rails...

**Comment by banesullivan** [comment]
It appears #1612 is the culprit for a lot of this stuff. @akaszynski, why did you add `return self` anytime `inplace=True`. I'm currently *very* opposed to this.

**Comment by banesullivan** [comment]
This was resolved by #1860
